app/controllers/patients_controller.rb

```diff
@@ -391,15 +391,32 @@ def update_fields(patient, params)
     patient.save
   end
 
-  def reset_symptom_onset(params_to_update, params, patient)
+  def reset_symptom_onset(content, patient, initiator)
     # Set user-defined symptom onset to be false and set the symptom onset date based on latest symptomatic report
-    params_to_update.concat(%i[user_defined_symptom_onset symptom_onset])
-    params[:patient][:user_defined_symptom_onset] = false
-    params[:patient][:symptom_onset] = patient.assessments.where(symptomatic: true).minimum(:created_at)
+    content[:user_defined_symptom_onset] = false
+    content[:symptom_onset] = patient.assessments.where(symptomatic: true).minimum(:created_at)
+
+    # Log system onset change in history if initiated by system (user initiated changes are logged separately)
+    return if content[:symptom_onset] == patient[:symptom_onset] || initiator != :system
+
+    comment = if !patient[:symptom_onset].nil? && !content[:symptom_onset].nil?
+                "System changed symptom onset date from #{patient[:symptom_onset].strftime('%m/%d/%Y')} to #{content[:symptom_onset].strftime('%m/%d/%Y')}
+                 because monitoree was moved from isolation to exposure workflow. This allows the system to show monitoree on appropriate line list based on
+                 daily reports."
+              elsif patient[:symptom_onset].nil? && !content[:symptom_onset].nil?
+                "System changed symptom onset date from blank to #{content[:symptom_onset].strftime('%m/%d/%Y')} because monitoree was moved from isolation to
+                 exposure workflow. This allows the system to show monitoree on appropriate line list based on daily reports."
+              elsif !patient[:symptom_onset].nil? && content[:symptom_onset].nil?
+                "System cleared symptom onset date from #{patient[:symptom_onset].strftime('%m/%d/%Y')} to blank because monitoree was moved from isolation to
+                 exposure workflow. This allows the system to show monitoree on appropriate line list based on daily reports."
+              else
+                'System changed symptom onset date. This allows the system to show monitoree on appropriate line list based on daily reports.'
+              end
+    History.monitoring_change(patient: patient, created_by: 'Sara Alert System', comment: comment)
   end
 
   def update_history(patient, params)
-    comment = 'User changed '
+    comment = ''
     comment += params.permit(:message)[:message] unless params.permit(:message)[:message].blank?
```

Comment:
I'd say combine these two lines

Reply:
addressed in https://github.com/SaraAlert/SaraAlert/pull/371/commits/c7d1425ada06125a87e6549ec62bdf34d4c380f2